ksa_compliance/utils/update_itemised_tax_data.py

```diff
@@ -36,16 +36,23 @@ def determine_if_export(doc):
         return False
 
     is_export = determine_if_export(doc)
-
+    included_in_print_rate = any(tax.included_in_print_rate for tax in doc.get("taxes", []))
     for row in doc.items:
         tax_rate, tax_amount = 0.0, 0.0
         # dont even bother checking in item tax template as it contains both input and output accounts - double the tax rate
         item_code = row.item_code or row.item_name
         if itemised_tax.get(item_code):
             for tax in itemised_tax.get(item_code).values():
                 _tax_rate = flt(tax.get("tax_rate", 0), row.precision("tax_rate"))
-                tax_amount += flt((row.net_amount * _tax_rate) / 100, row.precision("tax_amount"))
                 tax_rate += _tax_rate
+                if included_in_print_rate:
+                    amount = flt(row.amount, row.precision("amount"))
+                    net_from_gross = amount / (1 + (_tax_rate / 100))
+                    tax_amount += flt(amount - net_from_gross, row.precision("tax_amount"))
```

Comment:
be sure after calculating both that the total equal to the original value if it is not calculate difference and remove it from net ammount